addes new lookup plugin ansible-template

privateip opened the pull request:
The new ansible-template lookup plugin provides a directive driven
approach to templating key/value pairs (facts) to another text format.
It is based on the Ansible playbook directive language.

Commit:
addes new lookup plugin ansible-template

The new ansible-template lookup plugin provides a directive driven
approach to templating key/value pairs (facts) to another text format.
It is based on the Ansible playbook directive language.

Commit:
fixup pep8 and boilerplate issues

gundalow commented on lib/ansible/plugins/lookup/ansible_template.py:
```diff
@@ -0,0 +1,316 @@
+        config_lines = list()
+
+        for item in spec:
+            name = item.get('name')
```
Do we need something to warn/error if unknown keywords are specified?

gundalow commented on lib/ansible/plugins/lookup/ansible_template.py:
```diff
@@ -0,0 +1,316 @@
+                if loop:
+                    loop_data = self.template("{{ %s }}" % loop, template_vars, fail_on_undefined=False)
+                    if not loop_data:
+                        display.warning("block '%s' skipped due to missing loop var '%s'" % (name, loop))
```
warn or error?

gundalow commented on lib/ansible/plugins/lookup/ansible_template.py:
```diff
@@ -0,0 +1,316 @@
+      description: Relative or absolute path to the template file.
+      required: True
+    root_fact:
+      description: Roots the template values at the root fact if specified
```
Detail why this may be useful

gundalow reviewed (commented):
General thoughts
1) Should the `name:` end up as a section comment in the produced text?
2) Could you please link to a valid template so I can test and have a play locally

Commit:
fix one bug and change lines to template

This change fixes a bug where the block would not properly identify the
template lines.

This change also changes the `lines` directive to `template` which is a
little more descriptive of its function.

gundalow commented:
Moved to https://github.com/ansible/ansible/pull/33229
